src/input/source/hidraw.rs

```diff
@@ -85,6 +94,16 @@ impl HidRawDevice {
                 let source_device = SourceDriver::new(composite_device, device, device_info);
                 Ok(Self::Fts3528Touchscreen(source_device))
             }
+            DriverType::XpadUhid => {
+                let device = XpadUhid::new(device_info.clone())?;
+                let source_device = SourceDriver::new(composite_device, device, device_info);
+                Ok(Self::XpadUhid(source_device))
+            }
+            DriverType::RogAlly => {
+                let device = RogAlly::new(device_info.clone())?;
+                let source_device = SourceDriver::new(composite_device, device, device_info);
+                Ok(Self::RogAlly(source_device))
+            }
```

Comment:
Since this device has no real input implementation, we should set a high polling rate to reduce the extra CPU overhead of running this as a source device:

```rust
let options = SourceDriverOptions {
    poll_rate: Duration::from_millis(500),
    buffer_size: 1024,
};
...
let source_device = SourceDriver::new_with_options(composite_device, device, device_info, options);
```